src/libraries/System.Security.Cryptography.ProtectedData/tests/SpanProtectedDataTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Security.Cryptography.ProtectedDataTests;
+
+public class SpanProtectedDataTests : ProtectedDataTests
+{
+    protected override byte[] Protect(byte[] plain, byte[]? entropy, DataProtectionScope scope)
+    {
+        ReadOnlySpan<byte> inputSpan = plain;
+        ReadOnlySpan<byte> entropySpan = entropy;
+        Span<byte> destination = stackalloc byte[400];
+        int written = ProtectedData.Protect(inputSpan, scope, destination, entropySpan);
+        return destination.Slice(0, written).ToArray();
+    }
+
+    protected override byte[] Unprotect(byte[] encrypted, byte[]? entropy, DataProtectionScope scope)
+    {
+        ReadOnlySpan<byte> inputSpan = encrypted;
+        ReadOnlySpan<byte> entropySpan = entropy;
+        Span<byte> destination = stackalloc byte[encrypted.Length];
+        int written = ProtectedData.Unprotect(inputSpan, scope, destination, entropySpan);
+        return destination.Slice(0, written).ToArray();
+    }
+
+    [Fact]
+    public void ZeroBufferThrows()
+    {
+        Assert.Throws<ArgumentException>(() =>
+            ProtectedData.Protect([1, 2, 3], DataProtectionScope.CurrentUser, Span<byte>.Empty));
```

Comment:
Instead of Assert.Throws, use AssertExtensions.Throws and test the parameter name.  Which I think is the second parameter.

```suggestion
        AssertExtensions.Throws<ArgumentException>(
            () => ProtectedData.Protect([1, 2, 3], DataProtectionScope.CurrentUser, Span<byte>.Empty),
            "destination");
```